---
layout: report
title: Seasonal influenza circulation patterns and projections for 2017-2018
linktitle: Sep 2017
categories:
 - reports
draft: true
---

<div class="row">
    <div class="col-md-3">
        <div class="io-container title">
            A/H3N2
        </div>
        <div class="spacer"></div>
    </div>
    <div class="col-md-9">

        <p><b>H3N2 continues to diversify with many coexisting clades, all of which carry several amino acid mutations at previously characterized epitopes.
        The majority of viruses fall into the 3c2.a clade which has been dominating globally for >3 years, but 3c3.a viruses continue to persist.
        The common ancestor of circulating H3N2 viruses is now more than 5 years old, which is rare for H3N2. Despite extensive genetic diversity, serological assays suggest limited antigenic evolution.
        We expect 3c2.a virus to continue to dominate and eventually take over.</b>

        <div class="spacer"></div>

        <p>We base our primary analysis on a set of viruses collected between Oct 2015 and Aug 2017, comprising >800 viruses per month in Dec 2016 to Mar 2017 and 100-200 viruses per month more recently.
        We use all available data when estimating frequencies of mutations and weight samples appropriately by regional population size and relative sampling intensity to arrive at a putatively unbiased global frequency estimate.
        Phylogenetic analyses are based on a representative sample of about 2000 viruses.

        {% embed_image /reports/figures/sep-2017/h3n2_counts.png 12 %}

        <p>At the level of major clades, 3c2.a has continued to dominate while 3c3.a persisted at about 20% frequency in Oceania and North America in the 2016-2017 season.
        Within 3c2.a, the clade 3c2.a1 comprising HA1 mutation N171K and HA2 mutations I77V and G155E had risen rapidly in early 2016 but later stabilized at about 50% with somewhat higher frequencies in Europe.

        {% embed_image /reports/figures/sep-2017/h3n2_clades.png 12 %}

        <p>We observe substantial diversification within the 3c2.a clade over the past year.
        There are 5 major subclades within 3c2.a that have reached appreciable frequency.
        These clades are characterized by the following mutations:
        <ol>
          <li>53N, 144R, 171K, 192T, 197H</li>
          <li>121K, 144K</li>
          <li>131K, 142K, 261Q</li>
          <li>135K, HA2:150E</li>
          <li>92R, 311Q</li>
        </ol>
        All of these clades possess mutations at previously characterized epitope sites.

        {% embed_image /reports/figures/sep-2017/h3n2_tree.png 12 %}

        All of these clades have risen in frequency and are now competing with one another.
        Clade 1 (53N, 144R, 171K, 192T, 197H) is highly derived in terms of sequence, but is at low frequency globally (currently less than 1%) despite making up the a substantial fraction of 2017 Oceania viruses.
        Clade 2 (121K, 144K) is beginning to decline in frequency and is now at ~21% global frequency.
        Clade 3 (131K, 142K, 261Q) and clade 4 (135K, HA2:150E) have each slowly climbed to approximately 20% frequency.
        However, clade 5 (92R, 311Q) has recently increased to 39% in global frequency.

        {% embed_image /reports/figures/sep-2017/h3n2_mutations_3c2a.png 12 %}

        All 5 of these clades appear somewhat drifted according to antigenic analysis (<a href="http://www.pnas.org/content/113/12/E1701">Neher et al 2016</a>) of HI measurements provided by the Influenza Division at the US Centers for Disease Control and Prevention.
        With antigenic differences estimated between 0.5 units and 2.0 units.

        {% embed_image /reports/figures/sep-2017/h3n2_tree_titer_model.png 12 %}

        The mutation T135K has come up in clade 2, 4, and 5 and is now globally at 40% frequency.
        The most rapidly growing clade carries mutations N121K/K92R/H311Q/E62G/R142G/T135K.
        This is a subclade of clade 5 above with addition of 62G, 142G and 135K.
        The model fit to the antigenic data estimates that this subclade of clade 5 has drifted by 2.2 antigenic units, 1 of which is associated with the E62G/R142G mutations and 1.2 with the T135K mutation (on this background).
        Note, however, that there is no evidence of antigenic evolution in the FRA data (one data point only).
        A small sister clade with mutation T135N (also part of clade 5) recently developed.
        Taken together, changes at position 135 likely have selective origin.

        {% embed_image /reports/figures/sep-2017/h3n2_tree_135k.png 12 %}

        <p>Clades 1,2, and 3 are part of 3c2.a but outside of the 3c2.a1 subclade.
        Clade 3 with T131K/R142K mutations continues to be common in China (but not in South Asia) with an embedded cluster of South and North American isolates from last summer.
        Subclusters within this clade have been expanding recently, but there is little evidence for antigenic evolution in the HI or FRA data.
        The 131K substitution is not seen outside this clade.

        <p>Clade 1 with substitutions N31S/D53N/S144R/N171K/I192T/Q197H was observed globally over the past 9 month but is now decreasing everywhere except Oceania
        HI data and the model fit suggest a moderate (4-fold, about 2 antigenic units) titer drop relative to A/HongKong/4801/2014.
        Despite the titer drop, this calde seems to be outcompeted almost everywhere.

        <p>The most notable subclade of 3c3.a is a clade carrying the mutation F193S.
        However, this clade along with its sister clades has been decreasing in recent month and it doesn't show a consistent signal of antigenic change.
        The F193D mutations is observed sporadically within the 3c2.a clade as well.

        {% embed_image /reports/figures/sep-2017/h3n2_3c2a.png 12 %}

        <p>In summary, we think that the clade most likely to increase further in frequency is the E62G/R142G/T135K clade.
        However, two other clades (T131K/R142K and 144K) have similar potential.
        The former has very rapidly growing subclade, the latter rose rapidly last year and has a slightly higher LBI than the other two.

    </div>
    <div class="col-md-3">
        <div class="io-container title">
            A/H1N1pdm
        </div>
        <div class="spacer"></div>
    </div>
    <div class="col-md-9">

        <p><b>A clade comprising mutations S74R and I295V has recently risen to >90% global frequency.
        Although it shows no antigenic distinction by ferret HI data, the rapidity of its rise suggests a selective origin.</b></p>

        <div class="spacer"></div>

        <p>We base our primary analysis on a set of viruses collected between Oct 2015 and Aug 2017, comprising approximately 100 viruses per month during relevant months of 2017.
        We use all available data when estimating frequencies of mutations and weight samples appropriately by regional population size and relative sampling intensity to arrive at a putatively unbiased global frequency estimate.
        Phylogenetic analyses are based on a representative sample of about 2000 viruses.

        {% embed_image /reports/figures/sep-2017/h1n1pdm_counts.png 12 %}

        <p>H1N1pdm recently underwent a selective sweep of the 6b.1 clade comprising mutations S162N and I216T.
        Nearly all currently circulating H1N1pdm viruses are clade 6b.1.
        Correspondingly, the H1N1pdm vaccine strain was updated in Sep 2016 to match 6b.1 viruses.

        <p>There has been little diversification within clade 6b.1 with a couple notable exceptions.
        There is a clade comprising mutations S74R and I295V that has been very successful.
        Within this clade, there is a more recent subclade with the additional mutation S164T.

        {% embed_image /reports/figures/sep-2017/h1n1pdm_tree.png 12 %}

        <p>The S74R and I295V clade has risen rapidly in frequency starting around Jan 2017.
        This clade has been very successful is now at approximately 92% global frequency.
        Within this clade the subclade comprising S164T has risen in frequency from April 2017 and is now at approximately 35% global frequency.

        <p>Other mutations have appeared within 6b.1 viruses including S183P, R205K and I324V, but none of these have been very successful and are now declining in response to the rise of S74R and I295V.

        {% embed_image /reports/figures/sep-2017/h1n1pdm_mutations.png 12 %}

        <p>Antigenic analysis using HI measurements provided by the Influenza Division at the US Centers for Disease Control and Prevention fails to show any distinction within recently circulating H1N1pdm viruses.
        Like the clades 6b and 6b.1 clade before it, the new S74R and I295V clade does not show any titer drop despite there being ample measurements.

        {% embed_image /reports/figures/sep-2017/h1n1pdm_tree_titer_model.png 12 %}

        It was previously observed that clade 6b viruses with mutation K163Q showed decreased titers in a subset of adult human sera due to immunodominance effects, despite continued high titers in ferret sera (<a href="http://www.pnas.org/content/111/44/15798">Linderman et al 2014</a>).
        Importantly, the new S74R and I295V clade has spread at a similarly rapid rate as did clades 6b and 6b.1.
        This rate of spread and displacement of existing viral diversity is consistent with a selective driver.

        {% embed_image /reports/figures/sep-2017/h1n1pdm_mutations_6b_6b1.png 12 %}

        <p><i>Given current patterns of growth and decline it is almost certain that the S74R and I295V clade will dominant in a years time.
        Although there is no antigenic effect visible via HI from ferret antisera, the rapidity of the rise of S74R and I295V suggests a selective origin.
        We would suggest a detailed examination of human serological data as it's already been established that human serology may differ from ferret serology in H1N1pdm viruses.</i>

    </div>

    <div class="col-md-3">
        <div class="io-container title">
            B/Vic
        </div>
        <div class="spacer"></div>
    </div>
    <div class="col-md-9">

        <p><b>A clade with a two amino acid deletion 162-/163- has altered serological properties and is increasing in frequency, albeit slowly.
        Two other clades (carrying mutations K209N and V87A/I175V) have increased in frequency moderately.</b></p>

        <div class="spacer"></div>

        <p>We base our primary analysis on a set of viruses collected between Oct 2015 and Aug 2017, comprising between 50 and 250 viruses per month during relevant months of 2017.
        We use all available data when estimating frequencies of mutations and weight samples appropriately by regional population size and relative sampling intensity to arrive at a putatively unbiased global frequency estimate.
        Phylogenetic analyses are based on a representative sample of about 2000 viruses.

        {% embed_image /reports/figures/sep-2017/vic_counts.png 12 %}

        <p>The mutation I117V has continued its slow ascent in frequency as is now >99% in the global population.
        Within this clade there is relatively little circulating genetic diversity.
        The only clades of note are a small clade with mutations I180V, D129G and deletions K162- and K163-, another small clade with mutations V87A and I175V and a third small clade with mutation K209N.

        {% embed_image /reports/figures/sep-2017/vic_tree.png 12 %}

        <p>The clade bearing 129G, 162–, 163– and 180V has remained at low frequency globally with current global estimates at ~4%.
        The clade bearing 87A and 175V has increased in frequency substantially in North America but remains low in the rest of the world.
        The clade bearing 209N has been more successful, approaching 15% globally and circulating in Asia.

        {% embed_image /reports/figures/sep-2017/vic_mutations.png 12 %}

        <p>The clade bearing 129G, 162–, 163– and 180V appears clearly drifted according to HI measurements provided by the Influenza Division at the US Centers for Disease Control and Prevention.
        We estimate an approximately 8-fold titer drop (3 antigenic units) of drift from vaccine strain B/Brisbane/60/2008.
        Note that a small number of recent viruses sampled in Hong Kong have a 9bp deletion overlapping codons 162, 163, 164, and 165 along with a mutation I180T.
        While based on limited data, the tree model estimates an 1.2 unit titer drop.

        {% embed_image /reports/figures/sep-2017/vic_tree_titer_model.png 12 %}

        <p><i>Although the clade bearing 129G, 162–, 163– and 180V is almost certainly antigenically drifted, it's slow spread up to the this point suggests that it is unlikely to predominate in the population in a year's time.
        We would advise careful monitoring as further mutations may enhance spread or similar mutations might emerge on fitter backgrounds.</i>

    </div>
    <div class="col-md-3">
        <div class="io-container title">
            B/Yam
        </div>
        <div class="spacer"></div>
    </div>
    <div class="col-md-9">

      <p><b>A clade comprising M251V with clade 3 viruses continues to dominate.
      The is little genetic differentiation within this clade and no evidence of antigenic evolution.</b>

      <div class="spacer"></div>

      <p>We base our primary analysis on a set of viruses collected between Oct 2015 and Aug 2017, comprising between 50 and 250 viruses per month during relevant months of 2017.
      We use all available data when estimating frequencies of mutations and weight samples appropriately by regional population size and relative sampling intensity to arrive at a putatively unbiased global frequency estimate.
      Phylogenetic analyses are based on a representative sample of about 2000 viruses.

      {% embed_image /reports/figures/sep-2017/yam_counts.png 12 %}

      Clade 2 viruses appear to be extinct from the world.
      Clade 3 viruses have continued to evolve and genetic diversity relative to vaccine strain B/Phuket/3073/2013 has emerged.
      Of note the clade comprising M251V dominates the population with over 99% frequency.
      The subclade within M251V comprising K211R has decreased in frequency and is now extinct or nearly extinct.
      The only notable subclade with the M251V clade is that bearing T76I that has circulated at low frequency.

      {% embed_image /reports/figures/sep-2017/yam_mutations.png 12 %}

      <p>Based on HI measurements provided by the Influenza Division at the US Centers for Disease Control and Prevention, we find no evidence for antigenic evolution within clade 3 viruses and the vaccine B/Phuket/3073/2013 remains well matched to circulating viruses.

      {% embed_image /reports/figures/sep-2017/yam_tree_titer_model.png 12 %}

      <p><i>Barring the emergence of a new selective variant we anticipate relative stasis of B/Yam viruses in the coming year.</i>

    </div>


</div>
